datadog/resource_datadog_monitor.go

```diff
@@ -274,9 +274,8 @@ func buildMonitorStruct(d BuiltResource) (*datadogV1.Monitor, *datadogV1.Monitor
 	if attr, ok := d.GetOk("notify_no_data"); ok {
 		o.SetNotifyNoData(attr.(bool))
 	}
-	if attr, ok := d.GetOk("new_host_delay"); ok {
-		o.SetNewHostDelay(int64(attr.(int)))
-	}
+	// Don't check with GetOk, doesn't work with 0
+	o.SetNewHostDelay(int64(d.Get("new_host_delay").(int)))
```

Comment:
What if new_host_delay isn't actually set ? This will error out no ?

Reply:
There is a default so it should always be set I think.

Reply:
Ah ok, do we have a test case for it to make sure  ?

Reply:
Yeah tests like TestAccDatadogMonitor_SilencedRemove ought to trigger this.